File: app/src/main/java/knf/kuma/queue/QueueAnimesAdapter.java
package knf.kuma.queue;

import android.app.Activity;
import android.preference.PreferenceManager;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import java.util.ArrayList;
import java.util.List;
import java.util.Locale;

import androidx.annotation.LayoutRes;
import androidx.annotation.NonNull;
import androidx.cardview.widget.CardView;
import androidx.recyclerview.widget.RecyclerView;
import butterknife.BindView;
import butterknife.ButterKnife;
import knf.kuma.R;
import knf.kuma.animeinfo.ActivityAnime;
import knf.kuma.commons.PatternUtil;
import knf.kuma.commons.PicassoSingle;
import knf.kuma.database.CacheDB;
import knf.kuma.pojos.QueueObject;

public class QueueAnimesAdapter extends RecyclerView.Adapter<QueueAnimesAdapter.AnimeHolder> {

    private Activity activity;
    private OnAnimeSelectedListener listener;
    private List<QueueObject> list = new ArrayList<>();

    QueueAnimesAdapter(Activity activity) {
        this.activity = activity;
        this.listener = (OnAnimeSelectedListener) activity;
    }

    @NonNull
    @Override
    public AnimeHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new AnimeHolder(LayoutInflater.from(parent.getContext()).inflate(getLayout(), parent, false));
    }

    @Override
    public void onBindViewHolder(@NonNull final AnimeHolder holder, int position) {
        final QueueObject object = list.get(position);
        final String img = PatternUtil.getCover(object.chapter.aid);
        PicassoSingle.get(activity).load(img).into(holder.imageView);
        holder.title.setText(object.chapter.name);
        int count = CacheDB.INSTANCE.queueDAO().countAlone(object.chapter.aid);
        holder.type.setText(String.format(Locale.getDefault(), count == 1 ? "%d episodio" : "%d episodios", count));
        holder.cardView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if (listener != null) listener.onSelect(object);
            }
        });
        holder.cardView.setOnLongClickListener(new View.OnLongClickListener() {
            @Override
            public boolean onLongClick(View v) {
                ActivityAnime.open(activity, object, img, holder.imageView);
                return true;
            }
        });
    }

    @Override
    public int getItemCount() {
        return list.size();
    }

    public void update(List<QueueObject> list) {
        this.list = list;
        notifyDataSetChanged();
    }

    public void clear() {
        listener = null;
    }

    @LayoutRes
    private int getLayout() {
        if (PreferenceManager.getDefaultSharedPreferences(activity).getString("lay_type", "0").equals("0")) {
            return R.layout.item_anim_queue;
        } else {
            return R.layout.item_anim_queue_grid;
        }
    }

    interface OnAnimeSelectedListener {
        void onSelect(QueueObject object);
    }

    class AnimeHolder extends RecyclerView.ViewHolder {
        @BindView(R.id.card)
        CardView cardView;
        @BindView(R.id.img)
        ImageView imageView;
        @BindView(R.id.title)
        TextView title;
        @BindView(R.id.type)
        TextView type;

        AnimeHolder(View itemView) {
            super(itemView);
            ButterKnife.bind(this, itemView);
        }
    }
}

File: app/src/main/java/knf/kuma/queue/QueueAnimesAdapter.kt
package knf.kuma.queue

import android.app.Activity
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.annotation.LayoutRes
import androidx.recyclerview.widget.RecyclerView
import knf.kuma.R
import knf.kuma.animeinfo.ActivityAnime
import knf.kuma.commons.DesignUtils
import knf.kuma.commons.PrefsUtil
import knf.kuma.commons.bind
import knf.kuma.commons.doOnUIGlobal
import knf.kuma.commons.load
import knf.kuma.commons.notSameContent
import knf.kuma.pojos.QueueObject
import java.util.Locale

internal class QueueAnimesAdapter internal constructor(private val activity: Activity) : RecyclerView.Adapter<QueueAnimesAdapter.AnimeHolder>() {
    private var listener: OnAnimeSelectedListener? = null
    private var list: MutableList<QueueObject> = ArrayList()

    private val layout: Int
        @LayoutRes
        get() = if (PrefsUtil.layType == "0")
            if (DesignUtils.isFlat) R.layout.item_anim_queue_material else R.layout.item_anim_queue
        else
            if (DesignUtils.isFlat) R.layout.item_anim_queue_grid_material else R.layout.item_anim_queue_grid

    init {
        this.listener = activity as OnAnimeSelectedListener
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): AnimeHolder {
        return AnimeHolder(LayoutInflater.from(parent.context).inflate(layout, parent, false))
    }

    override fun onBindViewHolder(holder: AnimeHolder, position: Int) {
        val queueObject = list[position]
        holder.imageView.load("https://cdn.animeav1.com/covers/${queueObject.chapter.aid}.jpg")
        holder.title.text = queueObject.chapter.name
        holder.type.text = String.format(Locale.getDefault(), if (queueObject.count == 1) "%d episodio" else "%d episodios", queueObject.count)
        holder.cardView.setOnClickListener { listener?.onSelect(queueObject) }
        holder.cardView.setOnLongClickListener {
            ActivityAnime.open(activity, queueObject, holder.imageView)
            true
        }
    }

    override fun getItemCount(): Int {
        return list.size
    }

    fun update(list: MutableList<QueueObject>) {
        if (this.list notSameContent list) {
            this.list = list
            doOnUIGlobal { notifyDataSetChanged() }
        }
    }

    fun clear() {
        listener = null
    }

    internal interface OnAnimeSelectedListener {
        fun onSelect(queueObject: QueueObject)
    }

    internal class AnimeHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val cardView: View by itemView.bind(R.id.card)
        val imageView: ImageView by itemView.bind(R.id.img)
        val title: TextView by itemView.bind(R.id.title)
        val type: TextView by itemView.bind(R.id.type)
    }
}
